cuqipy_fenics/geometry.py

```diff
@@ -398,3 +403,243 @@ def _build_basis(self):
         # Normalize the eigenvectors if required
         if self.normalize:
             self._eig_vec /= np.linalg.norm( self._eig_vec )
+
+
+# Helper user expression to define the step expansion basis
+class _StepExpression(dl.UserExpression):
+    def __init__(self, degree, x_lim, y_lim=None):
+        self.x_lim = x_lim
+        self.y_lim = y_lim
+        super().__init__(degree=degree)
+
+    def eval(self, value, x):
+        if (
+            x[0] >= self.x_lim[0]
+            and x[0] < self.x_lim[1]
+            and (self.y_lim is None or x[1] >= self.y_lim[0])
+            and (self.y_lim is None or x[1] < self.y_lim[1])
+        ):
+            value[0] = 1
+        else:
+            value[0] = 0
+
+
+class StepExpansion(_WrappedGeometry):
+    """A geometry class that parameterizes finite element functions on a 1D or
+    2D domain with piecewise constant functions. In the 1D case, the domain is
+    divided into `num_steps_x` constant functions, each function support is of
+    length `L_x`/`num_steps_x`, where `L_x` is the length of the domain in the
+    x direction. In the 2D case, the domain is divided into `num_steps_x`
+    :math:`\\times` `num_steps_y` constant functions each has a support area of
+    `L_x`/`num_steps_x` x `L_y`/`num_steps_y`, where `L_y` is the length of the
+    domain in the y direction. The constant functions are arranged in a row-wise
+    order, from the bottom to top of the domain. That is, the first `num_steps_x`
+    functions are in the bottom row, the next `num_steps_x` functions are in the
+    next row, and so on. The step expansion is built on the given input geometry
+    that specifies the computational mesh.
```

Comment:
This is really useful information. Could we create some paragraphs to make it easier to read? Like splitting for the 1D and 2D cases.

Reply:
Thank you @nabriis , this is a good point, I updated the documentation